### PR TITLE
Modify swap chain IDL to reflect consensus in the F2F

### LINKED ISSUE #182: What is the purpose of WebGPUSwapChain.present()?
We have agreed to maintain the invariant that nothing should show up on the screen until the end of a rAF(). That begs the question why the present() call exists in the first place. In WebGL, there is no present() call, and the results show up at the end of rAF().

This matters regarding GPUSwapChain.getNextTexture(). Do you get the same texture if you call this twice in the same rAF()?

### PR DESCRIPTION
**Pull request by RafaelCintron**
Added required 'context' and 'format' members to GPUSwapChainDescriptor dictionary
Added 'usage' member to GPUSwapChainDescriptor dictionary with a default value of OUTPUT_ATTACHMENT.
Renamed GPUSwapChain.getNextTexture to GPUSwapChain.getCurrentTexture
Removed GPUSwapChain.present. "Presenting" happens implicitly like WebGL does
Removed GPUSwapChain.configure. When canvas is resized, developer must make a new swapchain from the GPUDevice.
Added GPUDevice.getSwapChainPreferredFormat. Developers can query for the preferred format and potentially receive something besides r8g8b8a8-unorm.
Added GPUDevice.createSwapChain. Developers must make a swap chain for a canvas in order for its contents to appear on the page.
Added GPUCanvasContext
Removed GPURenderingContext

Fixes #182

### CODE REVIEW

File: design/sketch.webidl [hunk 1]
```diff
@@ -627,14 +627,13 @@ interface GPUQueue {
 
 // SwapChain / RenderingContext
 dictionary GPUSwapChainDescriptor {
-    GPUTextureUsageFlags usage;
-    GPUTextureFormat format;
+    required GPUCanvasContext context;
```
Comment (Kangz) [inline design/sketch.webidl]: Should `GPUCanvasContext` be defined somewhere in this IDL?

File: design/sketch.webidl [hunk 2]
```diff
@@ -627,14 +627,13 @@ interface GPUQueue {
...
 };
 
 interface GPUSwapChain {
-    void configure(GPUSwapChainDescriptor descriptor);
-    GPUTexture getNextTexture();
-    void present();
+    GPUTexture getTexture();
 };
 
 interface GPURenderingContext : GPUSwapChain {
```
Comment (kvark) [inline design/sketch.webidl]: we need to remove this

Comment (RafaelCintron) [reply design/sketch.webidl]: @kvark , can you please elaborate why you would like `getTexture` to be removed?  How else would the web developer determine where to draw?

File: design/sketch.webidl [hunk 3]
```diff
@@ -675,10 +674,16 @@ interface GPUDevice {
     GPUCommandBuffer createCommandBuffer(GPUCommandBufferDescriptor descriptor);
     GPUFence createFence(GPUFenceDescriptor descriptor);
 
+    // Calling createSwapChain a second time for the same GPUCanvasContext
+    // invalidates the previous one, and all of the textures it’s produced.
+    GPUSwapChain createSwapChain(GPUSwapChainDescriptor desc);
```
Comment (kainino0x) [inline design/sketch.webidl]: nit: Other create* functions use descriptor instead of desc

File: design/sketch.webidl [hunk 4]
```diff
@@ -625,19 +625,18 @@ interface GPUQueue {
...
-    void present();
+dictionary GPUSwapChainDescriptor {
+    required GPUCanvasContext context;
+    required GPUTextureFormat format;
+    GPUTextureUsageFlags usage = GPUTextureUsage.OUTPUT_ATTACHMENT;
 };
 
-interface GPURenderingContext : GPUSwapChain {
+interface GPUSwapChain {
+    GPUTexture getTexture();
```
Comment (kainino0x) [inline design/sketch.webidl]: bikeshed: I'm still partial to getCurrentTexture. getTexture makes it sound like there's only one texture for the life of the swapchain (although if a programmer makes this error, they will probably figure it out pretty fast).

**Comment by kainino0x** [comment]
I added "Fixes #182" to the summary of this PR, so it'll auto-close that issue when merged.

**Pull request merged**